src/PrimitiveObsession.Test/EngineContainerBuilderTest.cs

```diff
@@ -13,12 +12,7 @@ public class EngineContainerBuilderTest
 
         public EngineContainerBuilderTest()
         {
-            var collection = new NameValueCollection {
-                { "TireCount", "4" },
-                { "PistonCount", "6" }
-            };
-
-            _sut = new EngineContainerBuilder(collection).Build();
+            _sut = new EngineContainerBuilder().Build(tireCount: "4", pistonCount: "6");
```

Comment:
Ok, qui scommetto che non ti piace: ho tolto il service locator da `EngineContainerBuilder`, che adesso riceve i due parametri direttamente, in modo che siano espliciti e visibili già leggendo la firma.
La cosa che potrebbe non piacerti è che non inietto più i parametri dal costruttore, ma direttamente nel metodo build: lo faccio perché `EngineContainerBuilder` è pensato per essere utilizzato una sola volta, per cui non gli fa per niente comodo avere delle variabili di istanza. Ma, soprattutto, inietto i valori da `Build()` perché così dopo mi appresto a fare l'in-line e eliminare del tutto il builder.